### PR TITLE
8285658: Fix two typos in the spec of j.u.random.RandomGenerator

### PR DESCRIPTION
**Pull request by rgiulietti**
The spec of the interface `java.util.random.RandomGenerator` is slightly incorrect when it discusses `float` and `double` random values.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8285658](https://bugs.openjdk.java.net/browse/JDK-8285658): Fix two typos in the spec of j.u.random.RandomGenerator
 * [JDK-8285641](https://bugs.openjdk.java.net/browse/JDK-8285641): Fix two typos in the spec of j.u.random.RandomGenerator (**CSR**)

### Reviewers
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**)
 * [Joe Darcy](https://openjdk.java.net/census#darcy) (@jddarcy - **Reviewer**) ⚠️ Review applies to 8d4388bd8ebfab0672e28709d0bad388f8d3ddf6

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8404/head:pull/8404` \
`$ git checkout pull/8404`

Update a local copy of the PR: \
`$ git checkout pull/8404` \
`$ git pull https://git.openjdk.java.net/jdk pull/8404/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8404`

View PR using the GUI difftool: \
`$ git pr show -t 8404`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8404.diff">https://git.openjdk.java.net/jdk/pull/8404.diff</a>

</details>

### CODE REVIEW

**Comment by rgiulietti** [comment]
This PR fixes the spec and replaces hard-coded literals with static compilation time symbolic expressions.

**Review by bplb — APPROVED** [review]

**Comment by liach** [comment]
Should we use `{@value }` javadoc tag, which renders the literal in HTML but includes a link to the constant?

File: src/java.base/share/classes/java/util/random/RandomGenerator.java [hunk 1]
```diff
@@ -496,11 +497,11 @@ default void nextBytes(byte[] bytes) {
      *
      * @return a pseudorandom {@code float} value between zero (inclusive) and one (exclusive)
      *
-     * @implSpec The default implementation uses the 24 high-order bits from a call to
-     * {@link RandomGenerator#nextInt() nextInt}().
+     * @implSpec The default implementation uses the {@link Float#PRECISION}
+     * high-order bits from a call to {@link RandomGenerator#nextInt() nextInt}().
```
Comment (jddarcy) [inline src/java.base/share/classes/java/util/random/RandomGenerator.java:501]: Typographical nit: please include the "()" inside the link tag for both methods.

**Review by jddarcy — APPROVED** [review]

**Pull request closed without merging**